internal/security/bootstrapper/command/setupacl/aclbootstrap.go

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package setupacl
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/edgexfoundry/go-mod-secrets/v2/pkg/token/fileioperformer"
+)
+
+// BootStrapACLTokenInfo is the key portion of the response metadata from consulACLBootstrapAPI
+type BootStrapACLTokenInfo struct {
+	AccessorID string   `json:"AccessorID"`
+	SecretID   string   `json:"SecretID"`
+	Policies   []Policy `json:"Policies"`
+	CreateTime string   `json:"CreateTime"`
+}
+
+// Policy is the metadata for ACL policy
+type Policy struct {
+	ID   string `json:"ID"`
+	Name string `json:"Name"`
+}
+
+// generateBootStrapACLToken should only be called once per Consul agent
```

Comment:
If each Consul agent has its own bootstrap token what does that mean when you have multiple Consul instances in the infrastructure?  Does that mean multiple bootstrap tokens?  Would a bootstrap token on one Consul instance be honored by another Consul instance?

Reply:
@bnevis-i : I googled it and from [Hashicorp's documentation](https://learn.hashicorp.com/tutorials/consul/access-control-setup-production#enable-acls-on-the-agents), the answers to your questions seem to be **yes** on `multiple tokens`: 

In order for ACL configuration to be applied correctly you will need to **apply the same parameters to every server and every client in your datacenter.** 

otherwise, you would think Consul should sync'ed up with one bootstrap token with all cluster automatically and internally, and no need to do it for every server and client in a datacenter.

Reply:
> In order for ACL configuration to be applied correctly you will need to **apply the same parameters to every server and every client in your datacenter.**

The way I read that was that the config.hcl needed to specify ACL mode on each Consul agent.

Later on it reads "To create the initial bootstrap token, use the acl bootstrap command _on one of the servers_."  (emphasis mine).  So it LOOKS like we're ok.

Reply:
ok, thanks for the details, good to know.